mypy/subtypes.py

```diff
@@ -143,10 +143,15 @@ def visit_instance(self, left: Instance) -> bool:
             rname = right.type.fullname()
             if not left.type.has_base(rname) and rname != 'builtins.object':
                 return False
-
             # Map left type to corresponding right instances.
             t = map_instance_to_supertype(left, right.type)
-
+            # Check that one of the types does not have type args or
+            # the number of type args is the same and
+            # each left type arg is acceptable in place of the matching right one
+            if not t.args or not right.args:
+                return True
+            if len(t.args) != len(right.args):
+                return False
```

Comment:
I'll look, but AFAIU `builtins.tuple[Any]` isn't the same as as `builtins.tuple`; the latter has an arbitrary number of arguments, while the former must have exactly 1. Are you saying that `mypy` should never create `Instance` that corresponds to `Tuple` (without type arguments)?